docs/github-actions-use-policy.md

```diff
@@ -0,0 +1,90 @@
+
+```yaml
+- name: Git Checkout
+  uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+```
+
+instead of
+
+```yaml
+- name: Git Checkout
+  uses: actions/checkout@v4
+```
+
+The use of SHAs can avoid security issues if the tag or branch of the action is moved to a malicious commit.
+We also gain build repeatability for future builds by referring to a precise revision of the action.
+
+Since we use Dependabot to manage our action dependencies, Dependabot will handle the chore of using the proper SHA values in the submitted pull requests when action dependencies are updated.
+
+## Trusted Providers of GitHub Actions
+
+There are many GitHub Actions available in GitHub.
+Not all can be necessarily trusted.
+The InstructLab project must maintain a list of allowed providers and a list of denied providers.
+
+Allowed providers will include all GitHub created actions (`actions/*`) as well as other official actions such as Python Packaging actions (`pypa/*`).
+We will need to inventory the current use of actions within the project's repositories to build out the list of allowed and denied providers.
+
+The InstructLab organization's Settings->Actions->General must be configured to allow select actions including actions created by GitHub along with the allowed providers.
+Adding additional actions to the allowed providers list will require approval by the organization maintainers along with updating the organization's settings.
+
+## Harden the GitHub Action runner
+
+To further increase security, the first step of each job in a workflow must be the [`step-security/harden-runner`](https://github.com/step-security/harden-runner) action.
+This action hardens the action runner by implementing network egress control and some infrastructure security.
+
+```yaml
+- name: "Harden Runner"
+  uses: step-security/harden-runner@a4aa98b93cab29d9b1101a6143fb8bce00e2eac4 # v2.7.1
+  with:
+    egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+```
+
+Each execution of the workflow job will produce a report showing endpoints accessed by the job and possible source file overwrites.
+The report will also suggest modifications to the harden-runner action's configuration to further increase security.
+
+```yaml
+- name: "Harden Runner"
+  uses: step-security/harden-runner@a4aa98b93cab29d9b1101a6143fb8bce00e2eac4 # v2.7.1
+  with:
+    disable-sudo: true
+    egress-policy: block
+    allowed-endpoints: >
+      github.com:443
+```
+
+Hardening the action runner like this can prevent malicious or misbehaving actions or the misuse of actions from exfiltrating secrets.
+The article [Harden-Runner Defends Against Arbitrary Command Execution in tj-actions/changed-files GitHub Action](https://www.stepsecurity.io/blog/defend-against-arbitrary-command-execution-in-tj-actions-changed-files) shows how the misuse of an action could allow an attacker with a well-crafted pull request to exfiltrate secrets.
```

Comment:
```suggestion
The article [Harden-Runner Defends Against Arbitrary Command Execution in `tj-actions/changed-files` GitHub Action](https://www.stepsecurity.io/blog/defend-against-arbitrary-command-execution-in-tj-actions-changed-files) shows how the misuse of an action could allow an attacker with a well-crafted pull request to exfiltrate secrets.
```

This would make it so `tj` doesn't need to be in spellcheck